vpr/src/device/rr_graph_view.h

```diff
@@ -248,6 +248,13 @@ class RRGraphView {
     inline const char* node_side_string(RRNodeId node) const {
         return node_storage_.node_side_string(node);
     }
+    /*@brief Get the switch used for the iedge'th edge from specified RRNodeId.*/
+    inline short edge_switch(RRNodeId id, t_edge_size iedge) const {
+        return node_storage_.edge_switch(id, iedge);
+    }
+    inline short edge_switch(const RREdgeId& edge) const {
+        return node_storage_.edge_switch(edge);
```

Comment:
For the ``edge_switch(const RREdgeId& edge)``, it is rarely used. See if we do need it or not. I think the first API is the recommended way to use. We should avoid to use the second way. 
The ``RREdgeId`` can be fully internal based on my past practice.

Reply:
If we do not make second  API here then we can not remove from rr_node_impl.h.  if you have any better idea I can follow that.

Reply:
@umariqbal-rs It is o.k. to keep it now. But please add code comments

```
It is recommended to use API edge_switch(RRNodeId id, t_edge_size iedge) rather than this one. It may be deprecated later.
```

Reply:
OK sure I will add that on Monday.